load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java

```diff
@@ -16,817 +16,240 @@
         }
-        if (cpsGcloudRubySubscriberCount > 0) {
-          Preconditions.checkArgument(cpsGcloudRubyPublisherCount > 0,
-              "--cps_gcloud_ruby_publisher must be > 0.");
-          clientParamsMap.put(
-              new ClientParams(ClientType.CPS_GCLOUD_RUBY_SUBSCRIBER, "gcloud-ruby-subscription" + i),
-              cpsGcloudRubySubscriberCount / cpsSubscriptionFanout);
-        }
-        if (cpsGcloudNodeSubscriberCount > 0) {
-          Preconditions.checkArgument(cpsGcloudNodePublisherCount > 0,
-              "--cps_gcloud_node_publisher must be > 0.");
-          clientParamsMap.put(
-              new ClientParams(ClientType.CPS_GCLOUD_NODE_SUBSCRIBER, "gcloud-node-subscription" + i),
-              cpsGcloudNodeSubscriberCount / cpsSubscriptionFanout);
-        }
-        if (cpsGcloudDotnetSubscriberCount > 0) {
-          Preconditions.checkArgument(cpsGcloudDotnetPublisherCount > 0,
-              "--cps_gcloud_dotnet_publisher must be > 0.");
-          clientParamsMap.put(
-              new ClientParams(ClientType.CPS_GCLOUD_DOTNET_SUBSCRIBER, "gcloud-dotnet-subscription" + i),
-              cpsGcloudDotnetSubscriberCount / cpsSubscriptionFanout);
-        }
-      }
-      // Set static variables.
-      Controller.resourceDirectory = resourceDirectory;
-      Client.messageSize = messageSize;
-      Client.loadtestDuration = loadtestDuration;
-      Client.publishBatchSize = publishBatchSize;
-      Client.maxMessagesPerPull = cpsMaxMessagesPerPull;
-      Client.pollDuration = kafkaPollDuration;
-      Client.broker = broker;
-      Client.zookeeperIpAddress = zookeeperIpAddress;
-      Client.requestRate = requestRate;
-      Client.maxOutstandingRequests = maxOutstandingRequests;
-      Client.numberOfMessages = numberOfMessages;
-      Client.burnInDuration = burnInDuration;
-      Client.publishBatchDuration = publishBatchDuration;
-      Client.partitions = partitions;
-      Client.replicationFactor = replicationFactor;
+    }
 
-      // Start a thread to poll and output results.
-      ScheduledExecutorService pollingExecutor = Executors.newSingleThreadScheduledExecutor();
-      if (verbose) {
-        pollingExecutor.scheduleWithFixedDelay(
-            () -> {
-              synchronized (pollingExecutor) {
-                printStats();
-              }
-            },
-            VERBOSE_STATS_POLL_SECONDS,
-            VERBOSE_STATS_POLL_SECONDS,
-            TimeUnit.SECONDS);
-      }
-      if (maxPublishLatencyTest) {
-        controller = controllerFunction.apply(project, clientParamsMap);
-        if (controller == null) {
-          System.exit(1);
-        }
-        runMaxPublishLatencyTest();
-      } else if (maxSubscriberThroughputTest) {
-        controller = controllerFunction.apply(project, clientParamsMap);
-        if (controller == null) {
-          System.exit(1);
+    private static class LanguageConverter extends EnumConverter<ClientType.Language> {
```

Comment:
Is there a reason that LanguageConverter is a private static class here, but the other converters reside in their own place? My inclination is to make them all private static classes like this one given the limited scope of their usage.

Reply:
LanguageConverter is a one liner that only has to exist at all because EnumConverter<ClientType.MessagingType>.class isn't valid java.  The only other converter that I can think of is TestParameterProviderConverter, which I have grouped with the test parameters.